src/action.spec.ts

```diff
@@ -0,0 +1,41 @@
+import { Action } from "./action";
+import { Step, SwapAction } from "./cnd/swaps_payload";
+import { Swap } from "./swap";
+
+const swapAction: SwapAction = { name: Step.Redeem, href: "/redeem" };
+
+describe("Action", () => {
+  it("has the same name than the swap action used to construct", () => {
+    const swap = (undefined as unknown) as Swap;
+    const action = new Action(swapAction, swap);
+
+    expect(action.name).toEqual(swapAction.name);
+  });
+
+  it("calls Swap.executeAction with the parameters of the swap action used to construct", async () => {
+    const mockExecuteAction = jest.fn(async () => {
+      return {};
+    });
+    const mockNextAction = jest.fn(async () => {
+      return swapAction;
+    });
+    const mockDoLedgerAction = jest.fn(async () => {
+      return "";
+    });
+    const swap = ({
+      executeAction: mockExecuteAction,
+      nextAction: mockNextAction,
+      doLedgerAction: mockDoLedgerAction
+    } as unknown) as Swap;
```

Comment:
It took me a while to see how you are controlling the test input (`swapAction`) here and how the setup is related to your assertions.

This test would be easier to read if the "arrange" part is only one or two statements.

Personally, I usually don't bother writing tests that need such a complicated setup because I am too lazy :sweat_smile: 
If the underlying code is still crucial and needs testing, there is probably a way of refactoring that logic out so it is less cumbersome to access.

Reply:
I'll try to do a follow-up PR.

Reply:
Not a biggy :)

Finishing RUA is more important I'd say.